README.md

```diff
@@ -1,45 +1,80 @@
-PiCo
+PiCo: High-Performance Data-Analytics Pipelines in C++
 ===============
-
-PiCo stands for **Pi**peline **Co**mposition.
-
-The main entity in PiCo is the *Pipeline*, basically a graph-composition of processing elements. 
-
-This model is intended to give the user an unique interface for both stream and batch processing, hiding completely data management and focusing only on operations, which are represented by Pipeline stages. 
-
-The DSL we propose is entirely implemented in C++11, exploiting the FastFlow library as runtime.
-
-## How to compile
-To build any PiCo application, the FastFlow library should be provided.
-FastFlow can be downloaded from the [GitHub repository](https://github.com/fastflow/fastflow) and then add a symbolic link to the FastFlow library into the PiCo directory:
-
-`$ ln -s /path/to/FastFlow/ff /PiCo/root/dir`
+[![Travis Build Status](https://travis-ci.org/alpha-unito/PiCo.svg?branch=master)](https://travis-ci.org/alpha-unito/PiCo)
+[![License: LGPL v3](https://img.shields.io/badge/License-LGPL%20v3-blue.svg)](https://www.gnu.org/licenses/lgpl-3.0)
+[![GitHub tag](https://img.shields.io/github/tag/alpha-unito/pico.svg)](http://github.com/alpha-unito/pico/releases)
+[![GitHub Issues](https://img.shields.io/github/issues/alpha-unito/pico.svg)](http://github.com/alpha-unito/pico/issues)
+
+PiCo (**Pi**peline **Co**mposition) is an open-source C++11 header-only DSL for *high-performance data analytics*, featuring low latency, high throughput,  and minimal memory footprint on multi-core platforms.
+
+Take a look to the [word-count](examples/word-count/pico_wc.cpp) code to see how easy is writing a PiCo pipeline!
+
+## Report bugs + get help
+https://github.com/alpha-unito/pico/issues/new
+
+## Build and run tests
+The following steps require `cmake >=3.1` as build system.
+
+Get PiCo:
+```bash
+git clone https://github.com/alpha-unito/PiCo.git pico
+```
+The current implementation is based on [FastFlow](https://github.com/fastflow/fastflow) as runtime system.
+Get it and a link it:
+```bash
+git clone https://github.com/fastflow/fastflow.git /path/to/fastflow
+ln -s /path/to/fastflow/ff pico/
+```
+:rescue_worker_helmet: A better solution for including FastFlow as dependency is under development!
+
+Build tests: (and examples)
+```bash
+cd pico
+mkdir build && cd build
+cmake .. -DPICO_ENABLE_UNIT_TEST=ON
+make
+```
+Run Tests:
+```bash
+cd tests
+sh run_tests.sh
```

Comment:
If I run sh run_tests.sh then it works fine, but if I run ./run_tests.sh then I get : 
./run_tests.sh: line 13:  2790 Segmentation fault      ./generate_lines dictionary.txt 1k > lines.txt

Reply:
That is 100% my fault, I have being seeing and ignoring the segfault since a while :) I will investigate.

Reply:
Fixed.